File: Day 1/1-12-2024_Part1.cpp
#include <iostream>
#include <fstream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    // Initialising empty vectors for the two columns
    vector<int> left;
    vector<int> right;
    ifstream file("/home/dell/CODING/Advent of Code 2024/Day 1/1-12-2024.txt");

    string line;
    while (getline(file, line)) {
        size_t spacePos = line.find(' ');
        if (spacePos != string::npos) {
            int left_number = stoi(line.substr(0, spacePos));
            int right_number = stoi(line.substr(spacePos + 1));

            // Appending the left side numbers in the left vector
            left.push_back(left_number);
            // Appending the right side numbers in the right vector
            right.push_back(right_number);
        }
    }
    file.close();

    int sum = 0;
    while (!left.empty()) {
        auto min_left_it = min_element(left.begin(), left.end());
        auto min_right_it = min_element(right.begin(), right.end());

        int diff = *min_left_it - *min_right_it;
        sum += abs(diff);

        // Removing the numbers after using them into the sum
        left.erase(min_left_it);
        right.erase(min_right_it);
    }

    cout << "Sum: " << sum << endl;
    return 0;
}

File: Day 1/1-12-2024_Part1.py
# Initialising empty lists for the two columns
left=[]
right=[]

with open('/home/dell/CODING/Advent of Code 2024/Day 1/1-12-2024.txt', 'r') as f:
    for line in f:
        # Appending the left side numbers in the left list
        left_number=line.split()[0]
        left.append(int(left_number))

        # Appending the right side numbers in the right list
        right_number=line.split()[1]
        right.append(int(right_number))

sum=0
while left!=[]:
    diff=min(left)-min(right)
    if diff>0:
        sum+=diff
    else:
        diff=-diff
        sum+=diff
    
    # Removing the numbers after using them into the sum
    left.remove(min(left))
    right.remove(min(right))
print("Sum:",sum)